src/citrine/jobs/waiting.py

```diff
@@ -128,16 +133,24 @@ def wait_while_executing(
         If fails to finish execution within timeout
 
     """
-    start = time()
+    start = time.time()
 
     def execution_is_finished():
-        status = execution.status()
-        if print_status_info:
-            _print_execution_status(status, start)
-        return not status.in_progress
-
-    while not execution_is_finished() and (time() - start < timeout):
-        sleep(interval)
+        if isinstance(execution, PredictorEvaluationExecution):
+            if collection is None:
+                raise ValueError("Must provide collection")
+            status = collection.get(execution.uid).status
+            if print_status_info:
+                _print_string_status(status, start)
+            return status != "INPROGRESS"
+        else:
```

Comment:
We have two different patterns for accessing execution status, one for predictor evaluations and one for the other workflows. Are we planning on transitioning everything over to the former type?

Reply:
yes, we're hoping to switch over for consistency.
I think the doc that we'd like to more towards is here: https://github.com/CitrineInformatics/ai-engine-documentation/blob/master/docs/status.md, although it looks like that doc has camel case strings, and PEW has all caps, so there's still work to be done.